[bug]: backtrace.c: stdbool.h needed for portability

### Severity

Major

### Versions

Git

### Components/Modules

backtrace.c

### Operating Environment

stdbool.h needs to be included 

For some reason, bfd_boolean didn't require stdbool.h but using bool directly requires it on some platforms.

### Frequency of Occurrence

Constant

### Issue Description

-

### Relevant log output

```shell
backtrace.c: In function 'process_section':
backtrace.c:130:9: error: unknown type name 'bool'
  130 |         bool line_found = 0;
      |         ^~~~
backtrace.c:74:1: note: 'bool' is defined in header '<stdbool.h>'; did you forget to '#include <stdbool.h>'?
   73 | #include <pthread.h>
  +++ |+#include <stdbool.h>
   74 | 
make[1]: *** [/usr/src/asterisk-22.10.1/Makefile.rules:157: backtrace.o] Error 1
make: *** [Makefile:402: main] Error 2
```

### Asterisk Issue Guidelines

- [x] Yes, I have read the Asterisk Issue Guidelines

--- a/main/backtrace.c
+++ b/main/backtrace.c
@@ -72,6 +72,8 @@
 
 #include <pthread.h>
 
+#include <stdbool.h>
+
 /* simple definition of S_OR so we don't have include strings.h */
 #define S_OR(a, b) (a && a[0] != '\0') ? a : b
 

